test/models/name_test.rb

```diff
@@ -2401,13 +2401,15 @@ def test_format_autonym
   def test_mysql_sort_order
     return unless sql_collates_accents?
 
-    # RuboCop gives false positives
+    # rubocop:disable Lint/UselessAssignment
+    # cop gives false positives
```

Comment:
If it's complaining about n2, n3, and n4, then I would not consider these to be false positives.  I recommend changing this to something like:
```
names = [
    create_test_name("Agaricus Aehou"),
    create_test_name("Agaricus Aeiou"),
    create_test_name("Agaricus Aeiøu"),
    create_test_name("Agaricus Aëiou"),
    create_test_name("Agaricus Aéiou"),
    create_test_name("Agaricus Aejou")]
names[4].update(author: "aÉIOU"))
x = Name.where(id: names.map(&:id)).order(:author).pluck(:author)
assert_equal(%w[Aehou Aeiou Aëiou aÉIOU Aeiøu Aejou], x)
```
(Note I haven't tested the above, but hopefully you get the idea)